pkg/api/injectsnippetsfs.go

```diff
@@ -0,0 +1,103 @@
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"html"
+	"io"
+	"io/fs"
+	"time"
+)
+
+type InjectSnippetsFS struct {
+	fs.FS
+	name    string
+	content string
+}
+
+func NewInjectIndexFS(fsys fs.FS, name string, marker string, snippets map[string]string) (fs.FS, error) {
+	if len(snippets) == 0 {
+		// no snippets, return the original marker
+		return fsys, nil
+	}
+
+	// read the content and inject snippets
+	f, err := fsys.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = f.Close() }()
+	all, err := io.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", name, err)
+	}
+
+	// inject snippets code into content
+	codeSnippets := renderCodeSnippets(snippets)
+	contentBytes := bytes.Replace(all, []byte(marker), codeSnippets, 1)
+	return &InjectSnippetsFS{
+		FS:      fsys,
+		name:    name,
+		content: string(contentBytes),
+	}, nil
+}
+
+func renderCodeSnippets(snippets map[string]string) []byte {
+	var b bytes.Buffer
+	for k, v := range snippets {
```

Comment:
This gives no way to order snippets.  That will give unexpected behaviour, and only some of the time.  I would prefer to have a JSON array of `{key: ..., value: ...}` objects; is that possible?